lib/rules/no-regex-spaces.js

```diff
@@ -37,11 +37,11 @@ module.exports = {
          * @private
          */
         function checkRegex(node, value, valueStart) {
-            const multipleSpacesRegex = /( {2,})+?/,
+            const multipleSpacesRegex = /( {2,})+?( [+*{?]|[^+*{?])/,
```

Comment:
Oh good catch. Though I think changing `[^+*{?]` to `[^ ]` is a mistake. In this case, we would still have the error described above:
`/  +/` gets converted to `/ {2}+/`

Additional comment:
The original code had `+?` on the end of the matching group. As far as I can tell it doesn't actually do anything. I noticed you moved the question mark into the group. I think this would cause the regex to never match more than 2 or 3 spaces in a row. Do you think it's fine if I remove the `+?` all together? i.e.:

`( {2,})( [+*{?]|[^+*{?]|$)`